core/dispute_coordinator/impl/sending_dispute.cpp

```diff
@@ -127,28 +134,46 @@ namespace kagome::dispute {
 
     has_failed_sends_ = false;
 
+    auto size = receivers.size();
+
+    asyncSendRequests(std::move(protocol), std::move(receivers));
+
+    SL_TRACE(logger_, "Requests dispatched. Sent {} requests", size);
+
+    return true;
+  }
+
+  void SendingDispute::asyncSendRequests(
+      std::shared_ptr<network::SendDisputeProtocol> &&protocol,
+      std::vector<std::tuple<primitives::AuthorityDiscoveryId,
+                             libp2p::peer::PeerId>> &&receivers) {
+    REINVOKE(*main_pool_handler_,
+             asyncSendRequests,
+             std::move(protocol),
+             std::move(receivers));
+
     for (auto &[authority_id, peer_id] : receivers) {
       deliveries_.emplace(authority_id, DeliveryStatus::Pending);
       protocol->doRequest(
```

Comment:
May hide main thread switch inside `RequestResponseProtocol::doRequest`

Reply:
Makes sense. @iceseer what do you think about it?